kouncil-frontend/libs/schema-registry/src/lib/avro/avro-utils.service.ts

```diff
@@ -0,0 +1,114 @@
+import {Injectable} from '@angular/core';
+import {
+  EnumType,
+  Field,
+  isArrayType,
+  isEnumType,
+  isMapType,
+  isRecordType,
+  RecordType
+} from "avro-typescript";
+import {ArrayType, BaseType, isUnion, MapType, NamedType} from "avro-typescript/lib/model";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AvroUtilsService {
+
+  private readonly MIN_RANDOM_NUMBER: number = 1;
+  private readonly MAX_RANDOM_NUMBER: number = 1000;
+
+  private readonly INTEGER_TYPES: string[] = ['int', 'long'];
+  private readonly NUMBER_TYPES: string[] = ['double', 'float'];
+  private readonly STRING_TYPES: string[] = ['string', 'fixed'];
+  private readonly BYTE_TYPE: string = 'bytes';
+  private readonly NULL_TYPE: string = 'null';
+  private readonly BOOLEAN_TYPE: string = 'boolean';
+
+  public fillAvroSchemaWithData(avroSchema: string): object {
+    const schema = JSON.parse(avroSchema) as RecordType;
+    let example = {};
+    schema.fields.forEach(field => {
+      this.processField(field, example);
+    });
+    return example;
+  }
+
+  private getRandomValueBasedOnType(type: string): number | string | boolean | null {
+    if (this.INTEGER_TYPES.includes(type)) {
+      return this.getRandomInt();
+    }
+    if (this.NUMBER_TYPES.includes(type)) {
+      return this.getRandomFloat();
+    }
+    if (this.STRING_TYPES.includes(type)) {
+      return this.getRandomString();
+    }
+    if (this.BYTE_TYPE === type) {
+      return this.getRandomBytes();
+    }
+    if (this.BOOLEAN_TYPE === type) {
+      return this.getRandomBoolean();
+    }
+    if (this.NULL_TYPE === type) {
+      return null;
+    }
+
+    return type;
+  }
+
+  private getRandomInt(): number {
```

Comment:
Can we move all of these methods to new (generators) service? Then we will have a pure avro utility class with `fillAvroSchemaWithData` and `processField`.